messaging/messaging_utils.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package messaging
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+func validateMessage(message *Message) error {
+	if message == nil {
+		return fmt.Errorf("message must not be nil")
+	}
+
+	target := bool2int(message.Token != "") + bool2int(message.Condition != "") + bool2int(message.Topic != "")
+	if target != 1 {
+		return fmt.Errorf("exactly one of token, topic or condition must be specified")
+	}
+
+	// validate topic
+	if message.Topic != "" {
+		if strings.HasPrefix(message.Topic, "/topics/") {
+			return fmt.Errorf("topic name must not contain the /topics/ prefix")
+		}
+		if !regexp.MustCompile("^[a-zA-Z0-9-_.~%]+$").MatchString(message.Topic) {
```

Comment:
Similarly here and elsewhere, maybe pull out the MustCompile to a var

Reply:
Done